specs/MultiProfile.md

```diff
@@ -363,17 +399,60 @@ interface ICoreWebView2Profile2 : ICoreWebView2Profile {
   [propget] HRESULT CookieManager([out, retval] ICoreWebView2CookieManager** cookieManager);
 }
 
-[uuid(1c1ae2cc-d5c2-ffe3-d3e7-7857035d23b7), object, pointer_default(unique)]
-interface ICoreWebView2Profile3 : ICoreWebView2Profile2 {
-  /// All webviews on this profile will be closed, and the profile will be marked for deletion.
-  /// After the Delete() call completes, The render process of webviews on this profile will
-  /// asynchronously exit with the reason:`COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED`.
-  /// See 'COREWEBVIEW2_PROCESS_FAILED_REASON::COREWEBVIEW2_PROCESS_FAILED_REASON_PROFILE_DELETED'
-  /// for more details. The profile directory on disk will be actually deleted when the browser
-  /// process exits. Webview2 creation will fail with the HRESULT is ERROR_INVALID_STATE(0x8007139FL)
-  /// if you create it with the same name as a profile that is being deleted.
-  HRESULT Delete(); 
+[uuid(2765B8BD-7C57-4B76-B8AA-1EC940FE92CC), object, pointer_default(unique)]
+interface ICoreWebView2Profile4 : IUnknown {
+  /// After the API is called, the profile will be marked for deletion. The
+  /// local profile's directory will be tried to delete at browser process
+  /// exit, if fail to delete, it will recursively try to delete at next
+  /// browser process start until successful.
+  /// The corresponding webview2s will be auto closed and its Closed event
+  /// handle function will be triggered with the reason is 
+  /// COREWEBVIEW2_CLOSED_REASON.COREWEBVIEW2_CLOSED_REASON_PROFILE_DELETED.
+  /// If create a new profile with the same name as the profile that has been
+  /// marked as deleted will be failure with the HRESULT:ERROR_INVALID_STATE
+  /// (0x8007139FL).
+  HRESULT Delete();
+}
+
+[uuid(cc39bea3-f6f8-471b-919f-fa253e2fff03), object, pointer_default(unique)]
+interface ICoreWebView2_9 : IUnknown {
+  /// Add an event handler for the `Closed` event. `Closed` enent handle runs
+  /// when the webview2 is closed passivly. When this event is raised, the 
+  /// webview2 cannot be used anymore.
+  HRESULT add_Closed(
+      [in] ICoreWebView2ClosedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_Closed`.
+  HRESULT remove_Closed(
+      [in] EventRegistrationToken token);
+}
+
+/// The reason of webview2 closed.
+[v1_enum]
+typedef enum COREWEBVIEW2_CLOSED_REASON {
+  /// Indicates that the reason of webview2 closed is its corresponding 
+  /// Profile has been or marked as deleted.
+  COREWEBVIEW2_CLOSED_REASON_PROFILE_DELETED,
+} COREWEBVIEW2_CLOSED_REASON;
```

Comment:
The two other cases we should add here (we can get the urgent work's implementation in first, and the less urgent parts implementation in a later PR) are for normal webview2 shutdown, and for browser process crash. Maybe this:
```
/// The CoreWebView2 has closed because the corresponding
/// CoreWebView2Controller had its Close method called either
/// explicitly or implicitly.
COREWEBVIEW2_CLOSED_REASON_SHUTDOWN,

/// The CoreWebView2 has closed because its browser process
/// crashed. The Closed event will be raised after the ProcessFailed
/// event is raised.
COREWEBVIEW2_CLOSED_REASON_BROWSER_PROCESS_FAILURE,
```

Reply:
Ok, I will create a work item to trace this things.

Reply:
Additional entries for this:

```
/// The CoreWebView2 has closed because the corresponding
/// CoreWebView2Controller had its Close method called either
/// explicitly or implicitly.
COREWEBVIEW2_CLOSED_REASON_SHUTDOWN,

/// The CoreWebView2 has closed because its browser process
/// crashed. The Closed event will be raised after the ProcessFailed
/// event is raised.
COREWEBVIEW2_CLOSED_REASON_BROWSER_PROCESS_FAILURE,

/// The CoreWebView2 has closed because its browser process
/// has exited. 
COREWEBVIEW2_CLOSED_REASON_BROWSER_PROCESS_EXITED,

/// The CoreWebView2 has closed because its parent window
/// was closed.
COREWEBVIEW2_CLOSED_REASON_PARENT_WINDOW_CLOSED
```

Reply:
Do I need to add those to the API SPEC, or add them one by one in a later PR?